### PR TITLE
[lint, type] Linting, docstring and type improvements in a*.py files

### PR DESCRIPTION
**Pull request by cidrblock**
depends-on: https://github.com/ansible/ansible-navigator/pull/671
- does what it says
- Additionally remove unused function (it got replaced by _argparse_error_handler in configurator long ago)

### CODE REVIEW

File: src/ansible_navigator/app.py [hunk 1]
```diff
@@ -78,46 +85,77 @@ def app(self) -> AppPublic:
...
 
     def parser_error(self, message: str) -> Tuple[None, None]:
-        """callback for parser error
+        """Store the argparse parser error.
+
+        This is used as a callback for argparse.
 
         :param message: A message from the parser
-        :type message: str
+        :return: Nothing
```
Comment (relrod) [inline src/ansible_navigator/app.py:136]: Is this method actually used?

```
main [ansible/winston] $ rg parser_error
src/ansible_navigator/app.py
41:        self._parser_error: str = ""
102:    def parser_error(self, message: str) -> Tuple[None, None]:
108:        self._parser_error = message
```

In any case, it's not returning `Nothing`, it's returning a two element tuple with both components being the value `None`.

File: src/ansible_navigator/app.py [hunk 2]
```diff
@@ -78,46 +84,67 @@ def app(self) -> AppPublic:
...
 
     @staticmethod
-    def _copy_args(args):
-        """deepcopy the args, but un mount the cache first
+    def _copy_args(args: ApplicationConfiguration) -> ApplicationConfiguration:
+        """Deepcopy the args.
+
+        Note: Unmount the collection doc cache (cdc) first
         the cdc will get mounted if the child needs it
         in parse and update
```
Comment (webknjaz) [inline src/ansible_navigator/app.py:92]: Let's improve the RST rendering a bit:
```suggestion
        .. note::

           Unmount the collection doc cache (cdc) first
           the cdc will get mounted if the child needs it
           in parse and update
```

File: src/ansible_navigator/action_runner.py [hunk 3]
```diff
@@ -1,16 +1,27 @@
...
 
 from .app import App
+from .configuration_subsystem import ApplicationConfiguration
 from .steps import Steps
 from .ui_framework import Interaction
 from .ui_framework import UIConfig
 from .ui_framework import UserInterface
 
+if TYPE_CHECKING:
+    from _curses import _CursesWindow  # pylint: disable= no-name-in-module
```
Comment (webknjaz) [inline src/ansible_navigator/action_runner.py:17]:
```suggestion
    from _curses import _CursesWindow  # pylint: disable=no-name-in-module
```

**Review by webknjaz — APPROVED** [review]

File: src/ansible_navigator/app.py [hunk 4]
```diff
@@ -78,46 +84,67 @@ def app(self) -> AppPublic:
...
         interaction.ui.scroll(0)
 
-    def parser_error(self, message: str) -> Tuple[None, None]:
-        """callback for parser error
-
-        :param message: A message from the parser
-        :type message: str
-        """
-        self._parser_error = message
-        return None, None
```
Comment (webknjaz) [inline src/ansible_navigator/app.py:129]:
```suggestion
```

**Pull request merged**